docs/reference/graphql.md

```diff
@@ -0,0 +1,170 @@
+
+## Usage
+
+To use this connector as a **connector client**, select the **GraphQL** tile when creating a new connector under **Search -> Connectors**.
+
+For additional operations, see [Usage](https://www.elastic.co/guide/en/enterprise-search/master/connectors-usage.html).
+
+## Configuration
+
+### Configure GraphQL connector
+
+Note the following configuration fields:
+
+#### `http_endpoint`  (required)
+
+Provide base URL of the source with endpoint. Example:
+
+ `https://api.xyz.com/graphql`
+
+#### `http_method`  (required)
+
+Select method from `GET` and `POST`.
+
+#### `authentication_method`  (required)
+
+Select authentication method from `No Auth`, `Basic Auth`, and `Bearer Token`.
+
+#### `username` (required in case of `Basic Auth`)
+
+In case of `Basic Auth` authentication method, provides username to connect with the source.
+
+#### `password` (required in case of `Basic Auth`)
+
+In case of `Basic Auth` authentication method, provides password to connect with the source.
+
+#### `token` (required in case of `Bearer Token`)
+
+In case of `Bearer token` authentication method, provides token to connect with the source.
+
+#### `graphql_query` (required)
+
+This query will be used to fetch data from the source. The query can contain variables which can be provided in the `graphql_variables` field. The connector will substitute the variables in the query with values from `graphql_variables` and make a GraphQL query to the source. Example:
+
+```
+query getUser($id: ID!) { 
+    user(id: $id) {
+        name 
+        email 
+    }
+}
+```
+
+#### `graphql_variables`
+
+A json of key value pairs of variables used in the GraphQL query. The connector will substitute the variables in the query with the values provided here and make a GraphQL query to the source.
+
+Example: If the GraphQL query is `query getUser($id: ID!) { user(id: $id) { name } }` and the value of `graphql_variables` is `{"id": "123"}`, the final query will be `query getUser { user(id: "123") { name } }` and the connector will make a GraphQL query to the source with the modified query.
+
+#### `graphql_object_list` (required)
+
+A JSON mapping between GraphQL response objects to index and their ID fields. The connector will fetch data for each object (JSON key) and use the provided ID field (JSON value) to index the object into Elasticsearch. The connector will index all fields for each object specified in the mapping. Use a dot `(.)` notation to specify the full path from the root of the GraphQL response to the desired object.
+
+Example: 
+
+```
+If the GraphQL query is `query getUser { organization { users{ user_id name email} } }` which is fetching all available users from the source and we want to index every user as separate document in that case configure this field as below.
```

Comment:
This comment should be outside of the code block, it will render as a one-line that is hard to read